bqskit/passes/synthesis/qsearch.py

```diff
@@ -232,3 +236,18 @@ async def synthesize(
             data['psols'] = psols
 
         return best_circ
+
+    def _set_layer_gen(self, data: PassData) -> LayerGenerator:
+        # Priority given to seeded synthesis
+        if 'seed_circuits' in data:
+            # Assumes that seed_circuits are compatible with machine model
+            layer_gen: LayerGenerator = SeedLayerGenerator(
+                seed=data['seed_circuits'],
```

Comment:
Can you send me an example where it breaks? data.gate_set should always be populated

Reply:
One of the QSearch tests was failing trying to run this. The test was using `circuit.perform` instead of `compiler.compile`. 
```
seed = Circuit(2)
seed.append_gate(CNOTGate(), (0, 1))

# Fails
data_dict = {'seed_circuits', [seed])
circuit.perform(qsearch, data_dict)

# Passes
update = UpdateDataPass('seed_circuits', [seed])
compiler.compile(circuit, [update, qsearch])
```

It looks like it happens because a `dict` is passed instead of a `PassData`. I've changed the default to be the `gate_set` defined `LayerGen`. Is there somewhere in `Circuit.perform` that the passed `data_dict` should be converted to a `PassData`?